spec/requests/v1/mixtape_request_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+describe 'mixtape endpoints' do
+  describe 'GET /mixtape' do
+    context 'with one genre specified' do
+      it 'responds with JSON for a mixtape' do
+        get(mixtape_url, { genre: 'blues' }, accept_headers)
+
+        expect(response).to have_http_status :ok
+        expect(response).to match_response_schema :mixtape
+      end
+    end
+    context 'with a genre and time in seconds specified' do
+      it 'responds with JSON for a mixtape' do
+        get(mixtape_url, { genre: 'blues', time: 3600 }, accept_headers)
+
+        expect(response).to have_http_status :ok
+        expect(response).to match_response_schema :mixtape
+      end
+    end
+  end
```

Comment:
You could also test these cases just in a test for the `MixtapeGenerator` service, I suppose.  Then here you could test that the `MixtapeGenerator.perform` method is called with the correct params.

Check out rspec-mocks for docs on how to test those situations: 

https://github.com/rspec/rspec-mocks

There's a few ways to do it but I usually do something like 

```ruby
allow(Service).to receive(:perform).with(genre: 'whatever', duration: 1000)

get(...)

expect(Service).to have_received(:perform).with(genre: 'whatever', duration: 1000)
```